service-course-crawler/src/main/kotlin/de/hpi/cloud/course/crawler/HpiCourseDetailPageCrawler.kt

```
@@ -0,0 +1,222 @@
+                                        .toMap()
+                                }
+                                RelevantSection.REQUIREMENTS -> {
+                                    requirements = elements.outerHtml()
+                                }
+                                RelevantSection.LEARNING -> {
+                                    learning = elements.outerHtml()
+                                }
+                                RelevantSection.EXAMINATION -> {
+                                    examination = elements.outerHtml()
+                                }
+                                RelevantSection.DATES -> {
+                                    dates = elements.outerHtml()
+                                }
+                                RelevantSection.LITERATURE -> {
+                                    literature = elements.outerHtml()
+                                }
+                                else -> {
+                                    elements.forEach { println(it.text()) }
+                                }
+                            }
+                        }
+                }
+            }
+        }
+    }
+
+    private fun parseDeadline(string: String): LocalDate? {
+        if(string.isBlank())
+            return null
+        try {
+            return LocalDate.parse(
+                string
+                    .replaceBefore("-", "")
+                    .trim('-', ' '),
+                SIMPLE_GERMAN_DATE_FORMAT
+            )
+        } catch (ex: DateTimeParseException) {
+            try {
+                return LocalDate.parse(
+                    string.split(" ")
+                        .subList(0, 3)
+                        .joinToString(separator = " "),
+                    LONG_GERMAN_DATE_FORMAT
+                )
+            } catch (ex: IndexOutOfBoundsException) {
+            } catch (ex: DateTimeParseException) {
+            }
```

Exceptions shouldn't be silently ignored - at least add a comment explaining the reason